src/GovUk.Education.ExploreEducationStatistics.Content.Model/Repository/ReleaseRepository.cs

```diff
@@ -51,24 +51,150 @@ await _contentDbContext.Entry(release)
+    {
+        var publication = await _contentDbContext.Publications
+            .Include(p => p.LatestPublishedRelease)
+            .SingleOrDefaultAsync(p => p.Id == publicationId);
+        return publication?.LatestPublishedRelease;
+    }
+
+    public async Task<Release?> GetLatestReleaseVersion(Guid publicationId)
+    {
+        var releases = await ListLatestReleaseVersions(publicationId);
+        return releases.FirstOrDefault();
+    }
+
+    public async Task<Release?> GetLatestPublishedReleaseVersion(Guid publicationId, string releaseSlug)
+    {
+        var releases = await _contentDbContext.Releases
+            .Where(release => release.PublicationId == publicationId
+                              && release.Slug == releaseSlug)
+            .ToListAsync();
+
+        return releases.SingleOrDefault(release => release.Live
+                                                   // It must also be the latest version unless the later version is a draft
+                                                   && !releases.Any(r =>
+                                                       r.Live
+                                                       && r.PreviousVersionId == release.Id
+                                                       && r.Id != release.Id));
+    }
+
+    public async Task<bool> IsLatestPublishedReleaseVersion(Guid releaseId)
     {
         var release = await _contentDbContext.Releases
-            .SingleAsync(r => r.Id == releaseId);
+            .SingleOrDefaultAsync(r => r.Id == releaseId);
+
+        if (release == null)
+        {
+            return false;
+        }
 
         var publication = await _contentDbContext.Publications
             .Include(p => p.Releases)
             .SingleAsync(p => p.Id == release.PublicationId);
 
-        return publication.IsLatestPublishedVersionOfRelease(release);
+        return
+            // Release itself must be live
+            release.Live
+            // It must also be the latest version unless the later version is a draft
+            && !publication.Releases.Any(r =>
+                r.Live
+                && r.PreviousVersionId == release.Id
+                && r.Id != release.Id);
     }
 
-    public async Task<Either<ActionResult, Release>> GetLatestPublishedRelease(Guid publicationId)
+    public async Task<bool> IsLatestReleaseVersion(Guid releaseId)
     {
+        var release = await _contentDbContext.Releases
+            .SingleOrDefaultAsync(r => r.Id == releaseId);
+
+        if (release == null)
+        {
+            return false;
+        }
+
         var publication = await _contentDbContext.Publications
-            .Include(p => p.LatestPublishedRelease)
-            .SingleOrDefaultAsync(p => p.Id == publicationId);
+            .Include(p => p.Releases)
+            .SingleAsync(p => p.Id == release.PublicationId);
+
+        return !publication.Releases.Any(r => r.PreviousVersionId == releaseId && r.Id != releaseId);
```

Comment:
Thanks, this is one which is quite simple to replace so I've gone ahead and done it. I've left the others as they are not as straight forward or would lead us fetching more data than we need to currently.